include/toolchain/gcc.h

```diff
@@ -274,7 +274,22 @@ do {                                                                    \
  * Need to use assembly macros because ';' is interpreted as the start of
  * a single line comment in the ARC assembler.
  */
+#if defined(__CCAC__)
```

Comment:
you probably want to create your own metaware.gcc and add special handling in there.

Reply:
Or just drop this entirely.   This whole mess is actually pre-existing, and is an attempt to fit the pre-existing GTEXT()/GDATA() assembly API for ARM and x86 into the ARC assembly syntax.

But... who cares?  It's an assembly macro.  You only every use it from ARC assembly.  There's zero value to making this portable at all, especially if it incurs this kind of messy syntax cost.

And it's a silly, trivial macro anyway.  All it does is declare an externally linked symbol.  Most of our more recent platforms (c.f. x86_64, xtensa don't use it at all, aarch64 does but only sparingly).  Really we've been moving away from most of our early assembly frameworks in a lot of our kernel APIs.  Maybe we should just junk this.  Now, for arc, would be  a good time.